src/Microphysics1M.jl

```diff
@@ -205,6 +205,149 @@ function terminal_velocity(
     return fall_w
 end
 
+# TODO - refactor
+"""
+    terminal_velocity(prs, precip, ρ, q_)
+
+ - `prs` - abstract set with Earth parameters
+ - `precip` - a type for rain or snow
+ - `spherical` - boolean that indicates spherical nature of particles
+ - `scheme` - type for parameterization
+ - `ρ` - air density
+ - `q_` - rain or snow specific humidity
+
+Returns the mass weighted average terminal velocity based on the formulas
+of Chen (2022). Integrated over the Marshall-Palmer (1948) distribution
+of rain drops and snow crystals.
+"""
+function terminal_velocity(
+    prs::APS,
+    precip::CT.AbstractPrecipType,
+    spherical::Bool,
+    scheme::CT.Chen2022Type,
+    ρ::FT,
+    q_::FT,
+) where {FT <: Real}
+    vt = FT(0)
+    if q_ > FT(0)
+
+        _n0::FT = n0(prs, q_, ρ, precip)
+        _r0::FT = r0(prs, precip)
+        _m0::FT = m0(prs, precip)
+        _me::FT = me(prs, precip)
+        _Δm::FT = Δm(prs, precip)
+        _a0::FT = a0(prs, precip)
+        _ae::FT = ae(prs, precip)
+        _Δa::FT = Δa(prs, precip)
+        _χm::FT = χm(prs, precip)
+        _χa::FT = χa(prs, precip)
+        _λ::FT = lambda(q_, ρ, _n0, _m0, _me, _r0, _χm, _Δm)
+        ρ_i::FT = CMP.ρ_cloud_ice(prs)
+        m0_comb::FT = _m0 * _χm
+        a0_comb::FT = _a0 * _χa
+        me_comb::FT = _me + _Δm
+        ae_comb::FT = _ae + _Δa
+
+        As = 0.00174079 * (log(ρ_i))^2 − 0.0378769 * log(ρ_i) - 0.263503
+        Bs = (0.575231 + 0.0909307 * log(ρ_i) + 0.515579 / sqrt(ρ_i))^(-1)
+        Cs =
+            -0.345387 +
+            0.177362 * exp(-0.000427794 * ρ_i) +
+            0.00419647 * sqrt(ρ_i)
+        Es = -0.156593 - 0.0189334 * (log(ρ_i))^2 + 0.1377817 * sqrt(ρ_i)
+        Fs = -exp(-3.35641 - 0.0156199 * (log(ρ_i))^2 + 0.765337 * log(ρ_i))
+        Gs =
+            (-0.0309715 + 1.55054 / (log(ρ_i)) - 0.518349 * log(ρ_i) / ρ_i)^(-1)
+
+        k = 3 #for mass weighted terminal velocity
+
+        if precip isa CT.RainType && spherical
+            ϕ = 1
```

Comment:
Definitely! Otherwise it's not type stable. I'll try to refactor the code a little to have the same function signature for all 1-moment terminal velocities. I can change all of those